theano/tensor/nnet/corr_gemm.c

```diff
@@ -203,22 +206,23 @@ PyArrayObject* corrMM(PyArrayObject* bottom,
     char Trans = 'T';
     PyArrayObject *output;
 
-    %(omp_set_threads)s(max_threads);
+    %(set_omp_threads)s(max_threads);
```

Comment:
Regarding Line 209, it's added in the first PR, this was added to make sure the parallelism codes running with "max_threads" and the omp_get_thread_num() to return correct omp index. Since this line will force the omp threads to `batch size` when `batch size` is smaller than `max_threads`, which will impact the other region of codes for example Line322 as you pointed out, besides codes without this line (209) are working as expected, so I think it's safe to remove this line. Then we can leave line 322 as it is and no need to restore omp threads at the end of OP.
Regarding the case of the computuer has more cores number than batch size, then the extra cores are not used as you said.